--- printer_ui-orig/printerui/qml/dialog/PrintFinish.qml	2026-05-19 21:11:16.148329771 -0400
+++ printer_ui/printerui/qml/dialog/PrintFinish.qml	2026-05-22 21:41:08.797480753 -0400
@@ -5,6 +5,7 @@
 
 import "qrc:/uibase/qml/widgets"
 import ".."
+import "../X1Plus.js" as X1Plus
 
 Item {
 
@@ -34,6 +35,15 @@
         }
 
         DialogButtonItem {
+            name: "print_again_usb"; title: qsTr("Print again")
+            visible: usbPrint !== null && !model.exists()
+            isDefault: isSelect === 0
+            onClicked: {
+                X1Plus.DDS.publish("device/request/print", usbPrint.payload)
+            }
+        }
+
+        DialogButtonItem {
             name: "report"; title: qsTr("Official Help")
             onClicked: {
                 dialogStack.popupDialog("ReportQRcode")
@@ -68,11 +78,13 @@
     property int isSelect: 0
     property var subtaskId
     property bool finish
+    property var usbPrint: null
 
     Component.onCompleted: {
         model = PrintManager.currentTask.model.clone()
         subtaskId = PrintManager.currentTask.getReportUrl().match(/taskId=(\S*)&startTime/)[1];
         finish = (PrintManager.currentTask.state == PrintTask.FINISH)
+        usbPrint = (typeof X1Plus.lastUsbPrint !== "undefined") ? X1Plus.lastUsbPrint : null
     }
 
     Component.onDestruction: {
